circuits/src/main.nr

```diff
@@ -0,0 +1,33 @@
+use dep::std;
+use dep::ecrecover;
+
+global CANDIDATES: Field = 2;
+
+fn main(
+  pub_key: [u8; 64],
+  signature: [u8; 64],
+  hashed_message: pub [u8; 32],
+  nullifier: pub [u8; 32],
+) {
+  let (pub_key_x, pub_key_y) = split_uncompressed_pub_key(pub_key);
+  
+  let address: Field = ecrecover::ecrecover(pub_key_x, pub_key_y, signature, hashed_message);
+  
+  let signature_hash = std::hash::blake2s(signature);
+  assert(nullifier == signature_hash);
```

Comment:
it was in the toy example but i'm not sure it's actually a good practice to check signature hash as nullifier

Reply:
Do you have idea about what should be nullifier?

Reply:
not really :/